Platform/Apple/tools/ProRWTS/prorwts2.s

```diff
@@ -145,1818 +218,3125 @@
+  !if (rwts_mode + enable_write) > 1 {
+                sta     unrcommand1 + 1
+  } ;rwts_mode = 1 and enable_write = 1
+  !if (rwts_mode + aligned_read + (enable_write xor 1)) = 0 {
+                sta     unrcommand2 + 1
+  } ;rwts_mode = 0 and aligned_read = 0 and enable_write = 1
+                sta     unrcommand3 + 1
+                lda     #>pcommand
+  !if (rwts_mode + enable_write) > 1 {
+                sta     unrcommand1 + 2
+  } ;rwts_mode = 1 and enable_write = 1
+  !if (rwts_mode + aligned_read + (enable_write xor 1)) = 0 {
+                sta     unrcommand2 + 2
+  } ;rwts_mode = 0 and aligned_read = 0 and enable_write = 1
+                sta     unrcommand3 + 2
+                lda     #$8e ;STX
+  !if (rwts_mode + aligned_read + (enable_write xor 1)) = 0 {
+                sta     unrcommand2
+  } ;rwts_mode = 0 and aligned_read = 0 and enable_write = 1
+                sta     unrbloklo
+                lda     #<pblock
+                sta     unrbloklo + 1
+                lda     #>pblock
+                sta     unrbloklo + 2
+                lda     #$8d ;STA
+                sta     unrblokhi
+                lda     #<(pblock + 1)
+                sta     unrblokhi + 1
+                lda     #>(pblock + 1)
+                sta     unrblokhi + 2
+                lda     #$a5 ;LDA
+                sta     unrunit1
+                lda     #adrlo
+                sta     unrunit1 + 1
+                lda     #$8d ;STA
+                sta     unrunit1 + 2
+                lda     #<paddr
+                sta     unrunit1 + 3
+                lda     #>paddr
+                sta     unrunit1 + 4
+
+                ;use SmartPort entrypoint instead
+
+                inx
+                inx
+                inx
+                stx     unrentry + 1
+
+                ldx     #2
+                stx     x80_parms + 4
+                lda     #0
+                sta     x80_parms + 5
+                jsr     MLI
+                !byte   $80
+                !word   x80_parms
+                lda     #cmdread
+                sta     unrpcommand
+                lda     #$ea
+                sta     hackstar
+
+iterunit        inc     unrunit2
+                jsr     unrentry
+                bcs     iterunit
+
++               ldy     #$10
+-               lda     readbuff + 3, y
+                cmp     readbuff + $203, y
+                bne     iterunit
+                dey
+                bne     -
+                lda     #$68
+                sta     hackstar
+                lda     #<packet
+                sta     unrppacket
+                lda     #>packet
+                sta     unrppacket + 1
+} ;use_smartport = 1
+
+bankram
+!if load_banked = 1 {
+                lda     LCBANK2 - ((lc_bank - 1) * 8)
+                lda     LCBANK2 - ((lc_bank - 1) * 8)
+} ;load_banked = 1
+!if load_aux = 1 {
+                sta     SETAUXWR + (load_banked * 4) ;SETAUXWR or SETAUXZP
```

Comment:
I think this is a bug: when load_aux and load_banked are both set, this line seems to set aux ZP. That in turn causes the plp 5 lines later to pull from uninitialized aux stack. In running this in Virtual II, it pulls $FF and causes the "omnidisk" in slot 7 to be treated a floppy.

Reply:
When I fix that, the initialization gets further though still crashes. I have to keep on tracing...